packages/client/src/task-queue-client.ts

```diff
@@ -0,0 +1,271 @@
+  public async getBuildIdCompatability(taskQueue: string): Promise<WorkerBuildIdVersionSets | undefined> {
+    let resp;
+    try {
+      resp = await this.workflowService.getWorkerBuildIdCompatibility({
+        taskQueue,
+        namespace: this.options.namespace,
+      });
+    } catch (e) {
+      this.rethrowGrpcError(e, 'Unexpected error fetching Build Id compatibility');
+    }
+    if (resp.majorVersionSets == null || resp.majorVersionSets.length === 0) {
+      return undefined;
+    }
+    return versionSetsFromProto(resp);
+  }
+
+  /**
+   * Fetches task reachability to determine whether a worker may be retired. The request may specify
+   * task queues to query for or let the server fetch all task queues mapped to the given build IDs.
+   *
+   * When requesting a large number of task queues or all task queues associated with the given
+   * build ids in a namespace, all task queues will be listed in the response but some of them may
+   * not contain reachability information due to a server enforced limit. When reaching the limit,
+   * task queues that reachability information could not be retrieved for will be marked with a
+   * `NotFetched` entry in {@link BuildIdReachability.taskQueueReachability}. The caller may issue
+   * another call to get the reachability for those task queues.
+   */
+  public async getBuildIdReachability(options: ReachabilityOptions): Promise<ReachabilityResponse> {
+    let resp;
+    try {
+      resp = await this.workflowService.getWorkerTaskReachability({
+        namespace: this.options.namespace,
+        taskQueues: options.taskQueues,
+        buildIds: options.buildIds,
+        reachability: reachabilityTypeToProto(options.reachability),
+      });
+    } catch (e) {
+      this.rethrowGrpcError(e, 'Unexpected error fetching Build Id reachability');
+    }
+    return reachabilityResponseFromProto(resp);
+  }
+
+  protected rethrowGrpcError(err: unknown, fallbackMessage: string): never {
+    if (isServerErrorResponse(err)) {
+      if (err.code === status.NOT_FOUND) {
+        throw new BuildIdNotFoundError(err.details ?? 'Build Id not found');
+      }
+      throw new ServiceError(fallbackMessage, { cause: err });
+    }
+    throw new ServiceError('Unexpected error while making gRPC request');
+  }
+}
+
+/**
+ * Options for {@link TaskQueueClient.getBuildIdReachability}
+ */
+export type ReachabilityOptions = RequireAtLeastOne<BaseReachabilityOptions, 'buildIds' | 'taskQueues'>;
+
+/**
+ * There are different types of reachability:
+ *   - `NewWorkflows`: The Build Id might be used by new workflows
+ *   - `ExistingWorkflows` The Build Id might be used by open workflows and/or closed workflows.
+ *   - `OpenWorkflows` The Build Id might be used by open workflows
+ *   - `ClosedWorkflows` The Build Id might be used by closed workflows
+ */
+export type ReachabilityType = 'NewWorkflows' | 'ExistingWorkflows' | 'OpenWorkflows' | 'ClosedWorkflows';
+
+interface BaseReachabilityOptions {
```

Comment:
Missing export
```suggestion
export interface BaseReachabilityOptions {
```

Reply:
This doesn't actually get used directly by users, only through the `ReachabilityOptions` type